edx_proctoring/tests/test_student_view.py

```diff
@@ -589,6 +590,116 @@ def test_get_studentview_submitted_timed_exam_with_past_due_date(self, due_date,
         else:
             self.assertIsNone(rendered_response)
 
+    @ddt.data(
+        (False, 'submitted', True, 1),
+        (True, 'verified', False, 1),
+        (False, 'submitted', True, 0),
+        (True, 'verified', False, 0),
+    )
+    @ddt.unpack
+    def test_get_studentview_submitted_timed_exam_with_grace_period(self, is_proctored, status, is_timed, graceperiod):
+        """
+        Test the student view for a submitted exam, after the
+        due date, when grace period is in effect.
+
+        Scenario: Given an exam with past due
+        When a user submission exists for that exam
+        Then get the user view with an active grace period
+        Then user will not be able to see exam content
+        And a banner will be visible
+        If the grace period is past due
+        For timed exam, user will not see any banner
+        And user will be able to see exam contents
+        And For proctored exam, view exam button will be visible
+        """
+        due_date = datetime.now(pytz.UTC) - timedelta(days=1)
+        context = {
+            'is_proctored': is_proctored,
+            'display_name': self.exam_name,
+            'default_time_limit_mins': 10,
+            'due_date': due_date,
+            'grace_period': timedelta(days=2)
+        }
+        exam_id = self._create_exam_with_due_time(
+            is_proctored=is_proctored, due_date=due_date
+        )
+        self._create_exam_attempt(exam_id, status=status)
+        rendered_response = get_student_view(
+            user_id=self.user_id,
+            course_id=self.course_id,
+            content_id=self.content_id_for_exam_with_due_date,
+            context=context
+        )
+        self.assertIn(self.wait_deadline_msg, rendered_response)
+
+        # This pop is required as the student view updates the
+        # context dict that was passed in the arguments
+        context.pop('wait_deadline')
+
+        context['grace_period'] = timedelta(days=graceperiod)
+        rendered_response = get_student_view(
+            user_id=self.user_id,
+            course_id=self.course_id,
+            content_id=self.content_id_for_exam_with_due_date,
+            context=context
+        )
+        if is_timed:
```

Comment:
Its not encouraging to have `if`/`else` in tests. Is there any particular reason why we can't have separate tests for timed exams?

Reply:
Well, the entire unit test is the same for both proctored/timed exam, except for the last condition.

Reply:
Right.

Reply:
I can separate them out(which I had done initially), but as I said that there was only a difference of one line, I combined them together.